nomad/server.go

```diff
@@ -1169,7 +1169,12 @@ func (s *Server) setupRaft() error {
 			}
 		} else if _, err := os.Stat(peersFile); err == nil {
 			s.logger.Info("found peers.json file, recovering Raft configuration...")
-			configuration, err := raft.ReadPeersJSON(peersFile)
+			var configuration raft.Configuration
+			if s.config.RaftConfig.ProtocolVersion < 3 {
+				configuration, err = raft.ReadPeersJSON(peersFile)
+			} else {
+				configuration, err = raft.ReadConfigJSON(peersFile)
```

Comment:
Now that I think about it - when running under version 3 - we may need to attempt to read in both formats.  I'll double check the raft track - but I suspect at version 2 - it starts writing peers file in new format.